cdp/wallet_address.py

```diff
@@ -221,6 +222,89 @@ def sign_payload(self, unsigned_payload: str) -> PayloadSignature:
             signature=signature,
         )
 
+    def deploy_token(
+        self, name: str, symbol: str, total_supply: Number | Decimal | str
+    ) -> SmartContract:
+        """Deploy a token smart contract.
+
+        Args:
+            name (str): The name of the token.
+            symbol (str): The symbol of the token.
+            total_supply (Union[Number, Decimal, str]): The total supply of the token.
+
+        Returns:
+            SmartContract: The deployed smart contract.
+
+        """
+        smart_contract = SmartContract.create(
+            wallet_id=self.wallet_id,
+            address_id=self.address_id,
+            type=SmartContract.Type.ERC20,
+            options=SmartContract.TokenContractOptions(
+                name=name, symbol=symbol, total_supply=str(total_supply)
```

Comment:
This can result in issues with different string representations of large `Decimal` values.

`amount=str(int(asset.to_atomic_amount(amount)))` - this is how we handle it for `Transfer.create(...)`